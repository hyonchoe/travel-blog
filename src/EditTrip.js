import React, { useState } from 'react'
import { Button, Input, DatePicker, Form, Space, Row, Col, Modal  } from 'antd'
import { CloseCircleOutlined } from '@ant-design/icons';
import './EditTrip.css'
import MyMapContainer from './MyMapContainer.js'


const EditTrip = props => {
    const [markerLatLng, setMarkerLatLng] = useState({
        lat: null,
        lng: null,
    })
    const [mapCenter, setMapCenter] = useState({
        // Default coordinate is New York, NY
        lat: 40.730610,
        lng: -73.935242,
    })
    const [addr, setAddr] = useState({
        city: '',
        state: '',
        country: '',
        fmtAddr: '',
    })
    const [disableDelBtns, setDisableDelBtns] = useState({
        loc0: true,
        loc1: true,
        loc2: true,
    })
    const [modalVisible, setModalVisible] = useState(false)        
    const [locFieldName, setLocFieldName] = useState('')

    const latLngDelim = ','
    const hiddenSuffix = '_hidden'
    const locations = {
        loc0: 'loc0',
        loc0Hidden: 'loc0' + hiddenSuffix,
        loc1: 'loc1',
        loc1Hidden: 'loc1' + hiddenSuffix,
        loc2: 'loc2',
        loc2Hidden: 'loc2' + hiddenSuffix,
    }
    /*
    const loc0 = 'loc0'
    const loc0Hidden = loc0 + hiddenSuffix
    const loc1 = 'loc1'
    const loc1Hidden = loc1 + hiddenSuffix
    const loc2 = 'loc2'
    const loc2Hidden = loc2 + hiddenSuffix
    */

    const getInitialMapCenter = () => {
        return {
            lat: 40.730610,
            lng: -73.935242,
        }
    }
    const getInitialAddr = () => {
        return {
            city: '',
            state: '',
            country: '',
            fmtAddr: '',
        }
    }
    const getInitialMarker = () => {
        return {
            lat: null,
            lng: null,
        }
    }

    const handleModalOk = () => {
        let value = {}
        value[locFieldName] = addr.fmtAddr
        let locFieldNameLatLng = locFieldName + hiddenSuffix
        value[locFieldNameLatLng] = markerLatLng.lat + latLngDelim + markerLatLng.lng
        form.setFieldsValue(value)

        //TODO
        let disableBtnsValue = {}
        disableBtnsValue[locFieldName] = (form.getFieldValue(locFieldName) === '')
        setDisableDelBtns({...disableDelBtns, ...disableBtnsValue})
        //
        setModalVisible(false)
        clearMapStates()

        setLocFieldName('')
    }
    const handleModalCancel = () => {
        setModalVisible(false)
        clearMapStates()

        setLocFieldName('')
    }
    const onLocSelected = (locAddrInfo, locLatLngInfo) => {
        setAddr(locAddrInfo)
        setMarkerLatLng(locLatLngInfo)
        setMapCenter(locLatLngInfo)
    }
    const clearLocation = (curLocFieldName) => {
        let reset = {}
        reset[curLocFieldName]=''
        form.setFieldsValue(reset)
        //TODO
        let disableBtnsValue = {}
        disableBtnsValue[curLocFieldName] = true
        setDisableDelBtns({...disableDelBtns, ...disableBtnsValue})
        //
    }
    const clearMapStates = () => {
        const resetAddr = getInitialAddr()
        const resetMarker = getInitialMarker()
        const resetMapCenter = getInitialMapCenter()

        setAddr(resetAddr)
        setMarkerLatLng(resetMarker)
        setMapCenter(resetMapCenter)
    }

    const onButtonClicked = (curLocFieldName) => {
        setModalVisible(true)
        setLocFieldName(curLocFieldName)
    }


    const [form] = Form.useForm()
    const existingTrip = props.editTrip
    let btnName = 'Submit'
    if (existingTrip){
        form.setFieldsValue({
            title: existingTrip.title,
            dates: [existingTrip.startDate, existingTrip.endDate],
            details: existingTrip.details,
        })
        btnName = 'Update'
    }

    const onFinish = values => {
        const tripData = {
            title: values.title,
            startDate: values.dates[0],
            endDate: values.dates[1],
            details: values.details,
            location: values.location,
        }

        const loc0FmtAddr = values[locations.loc0]
        const loc0FmtAddrHidden = values[locations.loc0Hidden]
        const loc1FmtAddr = values[locations.loc1]
        const loc1FmtAddrHidden = values[locations.loc1Hidden]
        const loc2FmtAddr = values[locations.loc2]
        const loc2FmtAddrHidden = values[locations.loc2Hidden]
        
        if (props.editTrip){
            props.handleUpdate(tripData, props.editTripId)
        }
        else{
            props.handleSubmit(tripData)
        }
    }
    const onCancel = () => {
        props.handleCancel()
    }

    return (
        <Row>
            <Col span={4} />
            <Col span={16}>
        <Form
            form={form}
            layout="vertical"
            onFinish={onFinish} >
            <Form.Item
                label="Trip Title"
                name="title"
                rules={[
                    {
                        required: true,
                        message: 'Please input your trip title.',
                    },
                ]}
                >
                <Input />
            </Form.Item>

            <Form.Item
                label="Date of Trip"
                name="dates"
                rules={[
                    {
                        required: true,
                        message: 'Please input your trip date.',
                    },
                ]} >
                <DatePicker.RangePicker
                    allowClear={true} />
            </Form.Item>
            
            <Form.Item
                label="Details"
                name="details"
                rules={[
                    {
                        required: true,
                        message: 'Pleaes input your trip details.',
                    },
                ]} >
                <Input.TextArea
                    autoSize={ {minRows:4, maxRows:20} } />
            </Form.Item>

            <Form.Item
                label="Location (up to three)" >
                <Form.Item
                    noStyle >                
                    <Form.Item
                        name={locations.loc0}
                        noStyle >
                        <Input readOnly={true} placeholder='Where did you go?' />
                    </Form.Item>
                    <CloseCircleOutlined
                        className="dynamic-delete-button"
                        disabled={disableDelBtns.loc0}
                        style={{ margin: '0 8px' }}
                        onClick={() => clearLocation(locations.loc0)} />
                    <Button type="link" onClick={() => onButtonClicked(locations.loc0)}>Select location</Button>                
                </Form.Item>
                <Form.Item 
                    name={locations.loc0Hidden}
                    noStyle
                    style={{ display: 'none' }} >
                    <Input readOnly={true} style={{ display: 'none' }} />
                </Form.Item>
                <Form.Item
                    noStyle >
                    <Form.Item
                        name={locations.loc1}
                        noStyle >
                        <Input readOnly={true} placeholder='Where did you go?' />
                    </Form.Item>
                    <CloseCircleOutlined
                        className="dynamic-delete-button"
                        disabled={disableDelBtns.loc1}
                        style={{ margin: '0 8px' }}
                        onClick={() => clearLocation(locations.loc1)}
                    />
                    <Button type="link" onClick={() => onButtonClicked(locations.loc1)}>Select location</Button>                
                </Form.Item>
                <Form.Item 
                    name={locations.loc1Hidden}
                    noStyle
                    style={{ display: 'none' }} >
                    <Input readOnly={true} style={{ display: 'none' }} />
                </Form.Item>
                <Form.Item
                    noStyle >
                    <Form.Item
                        name={locations.loc2}
                        noStyle >
                        <Input readOnly={true} placeholder='Where did you go?' />
                    </Form.Item>
                    <CloseCircleOutlined
                        className="dynamic-delete-button"
                        disabled={disableDelBtns.loc2}
                        style={{ margin: '0 8px' }}
                        onClick={() => clearLocation(locations.loc2)}
                    />
                    <Button type="link" onClick={() => onButtonClicked(locations.loc2)}>Select location</Button>                
                </Form.Item>
                <Form.Item 
                    name={locations.loc2Hidden}
                    noStyle
                    style={{ display: 'none' }} >
                    <Input readOnly={true} style={{ display: 'none' }} />
                </Form.Item>                

            </Form.Item>

            <Form.Item>
                    <Space>
                        <Button type="primary" htmlType="submit">{btnName}</Button>
                        {existingTrip && 
                            <Button type="link" onClick={onCancel}>Cancel</Button>
                        }
                    </Space>
            </Form.Item>
            <Col span={4} />
        </Form>
        </Col>

        <Modal
            title="Search for the trip location"
            visible={modalVisible}
            bodyStyle={{height: '550px'}}
            width="500px"
            maskClosable={false}
            onOk={handleModalOk}
            onCancel={handleModalCancel} >
            <MyMapContainer
                    searchMode={true}
                    tripLocations={null}
                    mapCenterLat={mapCenter.lat}
                    mapCenterLng={mapCenter.lng}
                    markerLat={markerLatLng.lat}
                    markerLng={markerLatLng.lng}
                    onLocSelected={onLocSelected} />
        </Modal>

        </Row>
    )
}

export default EditTrip