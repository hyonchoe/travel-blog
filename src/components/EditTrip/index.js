import React, { useState, useEffect } from 'react'
import { Button, Input, DatePicker, Checkbox, Form, Space, Spin, Row, Col, Typography  } from 'antd'
import { Prompt } from 'react-router-dom'
import history from '../../services/history'
import LocationSelect from '../LocationSelect'
import S3Upload from '../S3Upload'
import useTripCreateUpdate from './helpers/useTripCreateUpdate'

const EditTrip = props => {
    const [showNavPrompt, setShowNavPrompt] = useState(true)
    
    useEffect(() => {
        // For showing prompt on re-loading and closing window
        const handleUnload = (e) => {
            if (showNavPrompt) {
                e.preventDefault();
                e.returnValue = true;
            }
        }

        window.addEventListener('beforeunload', handleUnload)

        return () => {
            window.removeEventListener('beforeunload', handleUnload)
            props.clearEditTrip()
        }
    }, [])

    const { savingInProgress, createTrip, updateTrip } = useTripCreateUpdate()

    const titleMaxLength = 40
    const detailsMaxLength = 1000
    const listName = 'locationList'
    const latLngDelim = ','
    const uploadFldName = 'files'
    const msgColLayoutSides = {
        xs: { span: 0 },
        sm: { span: 2 },
        md: { span: 3 },
        lg: { span: 4 },
        xl: { span: 5 },
        xxl: { span: 6 }
    }
      const msgColLayoutContent = {
        xs: { span: 24 },
        sm: { span: 20 },
        md: { span: 18 },
        lg: { span: 16 },
        xl: { span: 14 },
        xxl: { span: 12 }
    }

    const formLayout = {
        labelCol: {
            sm: { span: 5 },
            md: { offset: 0, span: 5 },
            lg: { offset: 1, span: 5 },
            xl: { offset: 2, span: 5 },
            xxl: { offset: 3, span: 5 },
        },
        wrapperCol: {
            sm: { span: 19 },
            md: { span: 16 },
            lg: { span: 12 },
            xl: { span: 10 },
            xxl: { span: 8 },
        }
      }
    const formTailLayout = {
        wrapperCol: {
            sm: { offset: 5 , span: 19 },
            md: { offset: 5 , span: 16 },
            lg: { offset: 6, span: 12 },
            xl: { offset: 7, span: 10 },
            xxl: { offset: 8, span: 8 },
        }
    }

    const getInitialFormValues = (existingTrip) => {
        // Initial values for upload images are handled inside own component
        if (existingTrip){
            const initialLocations = (existingTrip.locations) ?
                                        existingTrip.locations.slice()
                                        : []
            initialLocations.forEach((loc) => {
                loc.latLng = loc.latLng[0] + latLngDelim + loc.latLng[1]
            })

            const initialValues = {
                title: existingTrip.title,
                dates: [existingTrip.startDate, existingTrip.endDate],
                details: existingTrip.details,
                public: existingTrip.public,
                [listName]: initialLocations,
            }
            
            return initialValues
        }

        return {}
    }

    const onFinish = values => {
        // Get location data
        const locationData = []
        const locationList = (values[listName]) ? values[listName] : []
        locationList.forEach((loc) => {
            if(loc.latLng){
                locationData.push({
                    fmtAddr: loc.fmtAddr,
                    latLng: loc.latLng.split(latLngDelim).map((coordinate) =>{
                        return parseFloat(coordinate)
                    }),
                    city: loc.city,
                    state: loc.state,
                    country: loc.country,
                })
            }
        })

        // Get image upload data
        const imageInfoData = []
        const uploadFiles = (values[uploadFldName]) ? values[uploadFldName] : []
        if (uploadFiles){
            if(uploadFiles.fileList){
                uploadFiles.fileList.forEach((file) => {
                    const nameToUrlName = uploadFiles.nameToUrlNameMap
                    if(nameToUrlName && nameToUrlName.hasOwnProperty(file.uid)){
                        const curFileNameInfo = nameToUrlName[file.uid]
                        imageInfoData.push({
                            name: curFileNameInfo.name,
                            fileUrlName: curFileNameInfo.fileUrlName,
                            S3Url: curFileNameInfo.pendingFileUrl,
                        })
                    }
                })
            }
        }

        const tripData = {
            title: values.title,
            startDate: values.dates[0].startOf('day'),
            endDate: values.dates[1].startOf('day'),
            public: (values.public) ? true : false,
            details: values.details,
            locations: locationData,
            images: imageInfoData,
        }
        if (props.editTrip){
            handleUpdate(tripData, props.editTrip._id)
        }
        else{
            handleSubmit(tripData)
        }
    }

    const onCancel = () => {
        props.clearEditTrip()
        history.push('/myTrips')
    }

    const handleSubmit = async (trip) => {
        const res = await createTrip(trip)
        if(res){
            setShowNavPrompt(false)
            history.push('/myTrips')
        }
    }
    const handleUpdate = async (updatedTrip, tripId) => {
        await updateTrip(updatedTrip, tripId)
        
        props.clearEditTrip()
        setShowNavPrompt(false)
        history.push('/myTrips')
    }
    
    const [form] = Form.useForm()
    const existingTrip = props.editTrip
    const existingImages = (existingTrip && existingTrip.images) ? existingTrip.images : []
    let btnName = 'Submit'
    let spinTip = 'Submitting the trip'
    if (existingTrip) {
        btnName = 'Update'
        spinTip = 'Updating the trip'
    }

    return (
        <Spin
            tip={spinTip}
            spinning={savingInProgress} >
            <Prompt
                when={showNavPrompt}
                message="You may have unsaved changes, are you sure you want to leave the page?" />
            <Row
                gutter={[0, 8]}
                justify="start" >
                <Col {...msgColLayoutSides} />
                <Col {...msgColLayoutContent}>
                    <Typography.Title>{greetingMsg((existingTrip) ? true:  false)}</Typography.Title>
                </Col>
                <Col {...msgColLayoutSides}/>
            </Row>
            <Form
                form={form}
                //{...layout}
                {...formLayout}
                initialValues={getInitialFormValues(existingTrip)}
                layout="horizontal"
                onFinish={onFinish} >
                    <Form.Item
                        label="Trip Title"
                        name="title"
                        rules={[
                            {
                                required: true,
                                message: 'Please input your trip title.',
                            },
                        ]} >
                        <Input maxLength={titleMaxLength} />
                    </Form.Item>

                    <Form.Item
                        label="Date of Trip"
                        name="dates"
                        rules={[
                            {
                                required: true,
                                message: 'Please input your trip date.',
                            },
                        ]} >
                        <DatePicker.RangePicker
                            allowClear={true} />
                    </Form.Item>

                    <Form.Item
                        //{...tailLayout}
                        {...formTailLayout}
                        name="public"
                        valuePropName="checked" >
                            <Checkbox>Allow public access</Checkbox>
                    </Form.Item>

                    <Form.Item
                        label="Details"
                        name="details"
                        rules={[
                            {
                                required: true,
                                message: 'Pleaes input your trip details.',
                            },
                        ]} >
                        <Input.TextArea
                            autoSize={ {minRows:4, maxRows:20} }
                            maxLength={detailsMaxLength} />
                    </Form.Item>

                    <LocationSelect
                        form={form}
                        listName={listName}
                        latLngDelim={latLngDelim} />

                    <S3Upload
                        form={form}
                        fieldName={uploadFldName}
                        images={existingImages} />

                    <Form.Item
                        //{...tailLayout}
                        {...formTailLayout} >
                            <Space>
                                <Button type="primary" htmlType="submit">{btnName}</Button>
                                <Button type="link" onClick={onCancel}>Cancel</Button>
                            </Space>
                    </Form.Item>
            </Form>
        </Spin>
    )
}

const greetingMsg = (editTrip) => {
    if (editTrip){
        return "What has changed?"
    }
    return "It's time to put one down in the books"
}

export default EditTrip