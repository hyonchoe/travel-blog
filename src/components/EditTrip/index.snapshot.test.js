import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'
import EditTrip from './'

describe('EditTrip', () => {
    let clearEditTrip
    beforeEach(() => {
        clearEditTrip = jest.fn()
    })

    it('matches snapshot for creating new trip', () => {

        const wrapper = shallow(
            <EditTrip
                editTrip={null}
                clearEditTrip={clearEditTrip} />
        )
        expect(wrapper).toMatchSnapshot()
    })

    it('matches snapshot for editing existing trip', () => {
        const dummyStartDate = moment('2020-07-01', 'YYYY-MM-DD')
        const dummyEndDate = moment('2020-07-04', 'YYYY-MM-DD')
        const dummyLocations = [
            { 
                fmtAddr: 'dummyfmtaddr1',
                latLng: [1, 1],
                city: 'dummycity1',
                state: 'dummystate1',
                country: 'dummycountry1',
            },
            { 
                fmtAddr: 'dummyfmtaddr2',
                latLng: [2, 2],
                city: 'dummycity2',
                state: 'dummystate2',
                country: 'dummycountry2',
            }
        ]
        const dummyImages = [
            {
                name: 'dummyname',
                fileUrlName: 'dummyurlname',
                S3Url: 'dummys3url'
            }
        ]
        const dummyTrip = {
            _id: 'dummyid',
            userId: 'dummyuserid',
            userName: 'dummyusername',
            userEmail: 'dummyuseremail',
            title: 'dummytitle',
            startDate: dummyStartDate,
            endDate: dummyEndDate,
            public: true,
            details: 'dummydetails',
            locations: dummyLocations,
            images: dummyImages,
        }
        const wrapper = shallow(
            <EditTrip
                editTrip={dummyTrip}
                clearEditTrip={clearEditTrip} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})