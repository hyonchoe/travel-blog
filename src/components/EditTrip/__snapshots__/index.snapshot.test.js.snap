// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTrip matches snapshot for creating new trip 1`] = `
<Spin
  size="default"
  spinning={false}
  tip="Submitting the trip"
  wrapperClassName=""
>
  <Prompt
    message="You may have unsaved changes, are you sure you want to leave the page?"
    when={true}
  />
  <Row
    gutter={
      Array [
        0,
        8,
      ]
    }
    justify="start"
  >
    <Col
      span={4}
    />
    <Col
      span={16}
    >
      <Title>
        It's time to put one down in the books
      </Title>
    </Col>
    <Col
      span={4}
    />
  </Row>
  <ForwardRef(InternalForm)
    form={
      Object {
        "__INTERNAL__": Object {},
        "getFieldError": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "getInternalHooks": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "resetFields": [Function],
        "scrollToField": [Function],
        "setFields": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
      }
    }
    initialValues={Object {}}
    labelCol={
      Object {
        "span": 8,
      }
    }
    layout="horizontal"
    onFinish={[Function]}
    wrapperCol={
      Object {
        "span": 10,
      }
    }
  >
    <FormItem
      label="Trip Title"
      name="title"
      rules={
        Array [
          Object {
            "message": "Please input your trip title.",
            "required": true,
          },
        ]
      }
    >
      <Input
        maxLength={40}
        type="text"
      />
    </FormItem>
    <FormItem
      label="Date of Trip"
      name="dates"
      rules={
        Array [
          Object {
            "message": "Please input your trip date.",
            "required": true,
          },
        ]
      }
    >
      <RangePicker
        allowClear={true}
      />
    </FormItem>
    <FormItem
      name="public"
      valuePropName="checked"
      wrapperCol={
        Object {
          "offset": 8,
          "span": 12,
        }
      }
    >
      <Checkbox
        indeterminate={false}
      >
        Allow public access
      </Checkbox>
    </FormItem>
    <FormItem
      label="Details"
      name="details"
      rules={
        Array [
          Object {
            "message": "Pleaes input your trip details.",
            "required": true,
          },
        ]
      }
    >
      <TextArea
        autoSize={
          Object {
            "maxRows": 20,
            "minRows": 4,
          }
        }
        maxLength={1000}
      />
    </FormItem>
    <LocationSelect
      form={
        Object {
          "__INTERNAL__": Object {},
          "getFieldError": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      latLngDelim=","
      listName="locationList"
    />
    <S3Upload
      fieldName="files"
      form={
        Object {
          "__INTERNAL__": Object {},
          "getFieldError": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      images={Array []}
    />
    <FormItem
      wrapperCol={
        Object {
          "offset": 8,
          "span": 12,
        }
      }
    >
      <Space>
        <Button
          block={false}
          ghost={false}
          htmlType="submit"
          loading={false}
          type="primary"
        >
          Submit
        </Button>
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="link"
        >
          Cancel
        </Button>
      </Space>
    </FormItem>
  </ForwardRef(InternalForm)>
</Spin>
`;

exports[`EditTrip matches snapshot for editing existing trip 1`] = `
<Spin
  size="default"
  spinning={false}
  tip="Updating the trip"
  wrapperClassName=""
>
  <Prompt
    message="You may have unsaved changes, are you sure you want to leave the page?"
    when={true}
  />
  <Row
    gutter={
      Array [
        0,
        8,
      ]
    }
    justify="start"
  >
    <Col
      span={4}
    />
    <Col
      span={16}
    >
      <Title>
        What has changed?
      </Title>
    </Col>
    <Col
      span={4}
    />
  </Row>
  <ForwardRef(InternalForm)
    form={
      Object {
        "__INTERNAL__": Object {},
        "getFieldError": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "getInternalHooks": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "resetFields": [Function],
        "scrollToField": [Function],
        "setFields": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
      }
    }
    initialValues={
      Object {
        "dates": Array [
          "2020-07-01T05:00:00.000Z",
          "2020-07-04T05:00:00.000Z",
        ],
        "details": "dummydetails",
        "locationList": Array [
          Object {
            "city": "dummycity1",
            "country": "dummycountry1",
            "fmtAddr": "dummyfmtaddr1",
            "latLng": "1,1",
            "state": "dummystate1",
          },
          Object {
            "city": "dummycity2",
            "country": "dummycountry2",
            "fmtAddr": "dummyfmtaddr2",
            "latLng": "2,2",
            "state": "dummystate2",
          },
        ],
        "public": true,
        "title": "dummytitle",
      }
    }
    labelCol={
      Object {
        "span": 8,
      }
    }
    layout="horizontal"
    onFinish={[Function]}
    wrapperCol={
      Object {
        "span": 10,
      }
    }
  >
    <FormItem
      label="Trip Title"
      name="title"
      rules={
        Array [
          Object {
            "message": "Please input your trip title.",
            "required": true,
          },
        ]
      }
    >
      <Input
        maxLength={40}
        type="text"
      />
    </FormItem>
    <FormItem
      label="Date of Trip"
      name="dates"
      rules={
        Array [
          Object {
            "message": "Please input your trip date.",
            "required": true,
          },
        ]
      }
    >
      <RangePicker
        allowClear={true}
      />
    </FormItem>
    <FormItem
      name="public"
      valuePropName="checked"
      wrapperCol={
        Object {
          "offset": 8,
          "span": 12,
        }
      }
    >
      <Checkbox
        indeterminate={false}
      >
        Allow public access
      </Checkbox>
    </FormItem>
    <FormItem
      label="Details"
      name="details"
      rules={
        Array [
          Object {
            "message": "Pleaes input your trip details.",
            "required": true,
          },
        ]
      }
    >
      <TextArea
        autoSize={
          Object {
            "maxRows": 20,
            "minRows": 4,
          }
        }
        maxLength={1000}
      />
    </FormItem>
    <LocationSelect
      form={
        Object {
          "__INTERNAL__": Object {},
          "getFieldError": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      latLngDelim=","
      listName="locationList"
    />
    <S3Upload
      fieldName="files"
      form={
        Object {
          "__INTERNAL__": Object {},
          "getFieldError": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      images={
        Array [
          Object {
            "S3Url": "dummys3url",
            "fileUrlName": "dummyurlname",
            "name": "dummyname",
          },
        ]
      }
    />
    <FormItem
      wrapperCol={
        Object {
          "offset": 8,
          "span": 12,
        }
      }
    >
      <Space>
        <Button
          block={false}
          ghost={false}
          htmlType="submit"
          loading={false}
          type="primary"
        >
          Update
        </Button>
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="link"
        >
          Cancel
        </Button>
      </Space>
    </FormItem>
  </ForwardRef(InternalForm)>
</Spin>
`;
